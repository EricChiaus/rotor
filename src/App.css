.App {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100vh;
}

.App > div {
    margin: 24px;
}
