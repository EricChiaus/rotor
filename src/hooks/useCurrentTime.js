import { useEffect, useState } from 'react';

const useCurrentTime = () => {
    const [date, setDate] = useState(new Date());

    useEffect(() => {
        const interval = setInterval(() => {
            setDate(new Date());
        }, 1000);
        return () => clearInterval(interval);
    }, []);

    return {
        hours: date.getHours(),
        minutes: date.getMinutes(),
        seconds: date.getSeconds(),
    };
};

export default useCurrentTime;
