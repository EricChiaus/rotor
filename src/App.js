import './App.css';
import useCurrentTime from './hooks/useCurrentTime';
import AnalogClockLayer from './components/AnalogClockLayer';
import DigitalClockLayer from './components/DigitalClockLayer';
import { TimeContext } from './contexts/TimeContext';

const App = () => {
    const time = useCurrentTime();

    return (
        <TimeContext.Provider value={time}>
            <div className="App">
                <AnalogClockLayer />
                <DigitalClockLayer />
            </div>
        </TimeContext.Provider>
    );
};

export default App;
