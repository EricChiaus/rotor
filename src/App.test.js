import { render } from '@testing-library/react';
import App from './App';
import useCurrentTime from './hooks/useCurrentTime';

jest.mock('./hooks/useCurrentTime', () => jest.fn());

describe('<App />', () => {
    beforeEach(() => {
        useCurrentTime.mockReset().mockReturnValue({
            hours: 0,
            minutes: 1,
            seconds: 59,
        });
    });
    it('matches the snapshot', () => {
        const { container } = render(<App />);
        expect(container).toMatchSnapshot();
    });
});
