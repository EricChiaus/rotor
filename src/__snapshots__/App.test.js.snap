// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="analog-clock"
    >
      <div
        class="hours"
        style="transform: rotateZ(0.5deg;"
      />
      <div
        class="minutes"
        style="transform: rotateZ(6deg;"
      />
      <div
        class="seconds"
        style="transform: rotateZ(354deg;"
      />
    </div>
    <div
      class="digital-clock"
    >
      <div
        class="zero digit"
      >
        <span
          class="d1"
        />
        <span
          class="d2"
        />
        <span
          class="d3"
        />
        <span
          class="d4"
        />
        <span
          class="d5"
        />
        <span
          class="d6"
        />
        <span
          class="d7"
        />
      </div>
      <div
        class="zero digit"
      >
        <span
          class="d1"
        />
        <span
          class="d2"
        />
        <span
          class="d3"
        />
        <span
          class="d4"
        />
        <span
          class="d5"
        />
        <span
          class="d6"
        />
        <span
          class="d7"
        />
      </div>
      <div
        class="dots"
      />
      <div
        class="zero digit"
      >
        <span
          class="d1"
        />
        <span
          class="d2"
        />
        <span
          class="d3"
        />
        <span
          class="d4"
        />
        <span
          class="d5"
        />
        <span
          class="d6"
        />
        <span
          class="d7"
        />
      </div>
      <div
        class="one digit"
      >
        <span
          class="d1"
        />
        <span
          class="d2"
        />
        <span
          class="d3"
        />
        <span
          class="d4"
        />
        <span
          class="d5"
        />
        <span
          class="d6"
        />
        <span
          class="d7"
        />
      </div>
      <div
        class="dots"
      />
      <div
        class="five digit"
      >
        <span
          class="d1"
        />
        <span
          class="d2"
        />
        <span
          class="d3"
        />
        <span
          class="d4"
        />
        <span
          class="d5"
        />
        <span
          class="d6"
        />
        <span
          class="d7"
        />
      </div>
      <div
        class="nine digit"
      >
        <span
          class="d1"
        />
        <span
          class="d2"
        />
        <span
          class="d3"
        />
        <span
          class="d4"
        />
        <span
          class="d5"
        />
        <span
          class="d6"
        />
        <span
          class="d7"
        />
      </div>
    </div>
  </div>
</div>
`;
